src/check_result.jl

```diff
@@ -56,19 +80,32 @@ function check_equal(
     @test ActualPrimal === ExpectedPrimal
 end
 
-# This catches comparisons of Composites and Tuples/NamedTuple
-# and gives a error messaage complaining about that
-check_equal(::C, expected::T) where {C<:Composite, T} = @test C === T
-check_equal(::T, expected::C) where {C<:Composite, T} = @test C === T
 
+# Some structual differential and a natural differential
```

Comment:
What are structural and natural differentials? Structural means Composite for a struct? Not sure about natural?

Reply:
natural is something a human decided made sense.
Structural just means `composite`
see https://www.juliadiff.org/ChainRulesCore.jl/dev/design/many_differentials.html